### PR TITLE
Add progress bar to delete device

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Add wait for delete device and progress bar during deleting

Fixes: 0

## Checklist
- [x] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [x] New code is covered with tests
- [x] Tests related to old code are updated
- [x] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 1]
```diff
@@ -31,6 +34,14 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
     }
   };
 
+  const onConfirmDeleteClick = async () => {
+    const currentLocation = history.location.pathname;
+    history.push(join(history.location.pathname, '/waitingForDeleteAccount'));
+    const {waitToBeSuccess} = await deployedWallet.removeKey(publicKey, transactionDetails);
+    await waitToBeSuccess();
+    history.replace(currentLocation);
```
Comment (rzadp) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: `history.goBack()` does this?

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 2]
```diff
@@ -43,7 +54,7 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
       {toBeRemoved
         ? <div className="connected-devices-buttons">
           <button onClick={() => setToBeRemoved(false)} className="connected-devices-cancel">Cancel</button>
-          <button onClick={() => deployedWallet.removeKey(publicKey, transactionDetails)} className="connected-devices-delete">Delete</button>
+          <button onClick={() => onConfirmDeleteClick()} className="connected-devices-delete">Delete</button>
```
Comment (rzadp) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]:
```suggestion
          <button onClick={onConfirmDeleteClick} className="connected-devices-delete">Delete</button>
```

File: universal-login-react/src/ui/UFlow/Devices/Devices.tsx [hunk 3]
```diff
@@ -44,6 +45,9 @@ export const Devices = ({walletService, onAccountDeleted, className, basePath =
           className={className}
         />
       </Route>
+      <Route path={`${basePath}/waitingForDeleteAccount`} exact>
```
Comment (rzadp) [inline universal-login-react/src/ui/UFlow/Devices/Devices.tsx]: Use `join`, like in other places - `basePath` can have trailing slash, and then there is a problem of `//`

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 4]
```diff
@@ -31,6 +34,14 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
     }
   };
 
+  const onConfirmDeleteClick = async () => {
+    const currentLocation = history.location.pathname;
+    history.push(join(history.location.pathname, '/waitingForDeleteAccount'));
```
Comment (Marik-D) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: In my opinion, using relative paths in navigation generally is an antipattern. It's really error prone and absolute paths are easier to reason about. Please pass `devicesBasePath` prop from parent component and use it to build a URL. Here's an example:
https://github.com/UniversalLogin/UniversalLoginSDK/blob/40d649f9a5d96edb6c8480dbca0b1aaac956dcb4/universal-login-react/src/ui/UFlow/Devices/Devices.tsx#L32
>

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx (hunk 1 above)
Comment (Marik-D) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/Devices.tsx (hunk 3 above)
Comment (Marik-D) [reply universal-login-react/src/ui/UFlow/Devices/Devices.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/Devices.tsx [hunk 5]
```diff
@@ -44,6 +45,9 @@ export const Devices = ({walletService, onAccountDeleted, className, basePath =
...
+        <WaitingFor action={'Deleting account'} className={className} />
```
Comment (Marik-D) [inline universal-login-react/src/ui/UFlow/Devices/Devices.tsx]:
```suggestion
        <WaitingFor action="Deleting account" className={className} />
```

File: universal-login-react/src/ui/UFlow/Devices/Devices.tsx (hunk 3 above)
Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/Devices.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx (hunk 2 above)
Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx (hunk 4 above)
Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/Devices.tsx (hunk 5 above)
Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/Devices.tsx]: 👍

**Pull request merged**